### PR TITLE
BI-2156 - Upgrade Apache Tika to 2.3.0+

### PR DESCRIPTION
**Pull request by mlm483**
# Description
**Story:** [BI-2156](https://breedinginsight.atlassian.net/browse/BI-2156)

- Upgraded apache tika to 2.9.2.
- Removed as many hard-coded transitive dependencies as possible.
- Hard-coded transitive dependency apache commons compress, because some tests were failing without specifying a compatible version.

# Testing
**You may need to do a maven `clean install`.**

- Test uploading files, verify no unexpected errors.
- Ensure tests pass.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have tested my code and ensured it meets the acceptance criteria of the story
- [ ] I have tested that my code works with both the brapi-java-server and BreedBase
- [ ] I have create/modified unit tests to cover this change
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to documentation
- [ ] I have run TAF: _\<please include a link to TAF run>_

[BI-2156]: https://breedinginsight.atlassian.net/browse/BI-2156?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -91,10 +91,9 @@
         <mockito.version>4.3.1</mockito.version>
         <brapi-java-client.version>2.1-SNAPSHOT</brapi-java-client.version>
         <commons-io.version>2.11.0</commons-io.version>
```
Comment (nickpalladino) [inline pom.xml:93]: Is the commons-io.version needed?

File: pom.xml [hunk 2]
```diff
@@ -378,36 +377,16 @@
             <artifactId>brapi-java-client</artifactId>
             <version>${brapi-java-client.version}</version>
         </dependency>
-        <dependency>
```
Comment (nickpalladino) [inline pom.xml:381]: I assume tika is pulling all these in?

**Review by nickpalladino — COMMENTED** [review]
Was able to import traits without error

File: pom.xml (hunk 1 above)
Comment (mlm483) [reply pom.xml]: It was unnecessary, I removed it. Thanks.

File: pom.xml (hunk 2 above)
Comment (mlm483) [reply pom.xml:381]: Yes, maven pulls those in to build tika as they are dependencies of tika. I don't know how common or advisable it is to hard-code transitive dependencies with maven projects, I used trial and error to determine that maven did the right thing for all the transitive dependencies besides apache commons compress, which I hard-coded. I think the issue might have been that two primary dependencies of bi-api both depend on apache commons compress, and maven was defaulting to using a version that didn't work with tika.

**Review by nickpalladino — APPROVED** [review]

**Review by davedrp — APPROVED** [review]
Passed Test

**Pull request merged**